### ISSUE TITLE
[Feature Request] Gestures

### ISSUE THREAD

**Issue by elichai** [nova-video-player/aos-AVP#34]
Hi,
This project looks really cool and I do love using open source projects, but in my opinion this app have one main drawback,
It doesn't support gestures, e.g. double tapping to pause, sliding horizontally to seek, sliding vertically to change brightness/volume etc.

Another great feature would be a notch support, to let users of Pixel 3 XL or other notch devices to get a real full screen experience.

Thanks!

**Comment by Dmitry-k42** [comment]
I would like to have gestures too. It is possible to make them customisable in settings. Who don't like it may disable it.

**Comment by Pentaphon** [comment]
I agree that gestures would be a great addition for touchscreen users and anybody who doesn't like it can just not enable them. The touchscreen UI is a bit cluttered the volume and progress bar sitting there like an axis. 

@courville any toughts on touchscreen gestures? I think they would add a modern touch to an otherwise aging touchscreen player UI.

**Comment by onthejobtraining** [comment]
Any updates on this must have features ?

**Comment by Ilithy** [comment]
Likewise, I love Nova, which I find perfect except for this missing feature, very practical on smartphones and tablets.
I am ready to support financially the development of this function if needed (I am not a coder, so I can't help in that)

Thank you very much for this wonderful app

**Comment by alexanderadam** [comment]
Would it be better to have a summary issue for all gestures or a dedicated issue for every gesture?

Maybe it would be helpful to have a structured list at least?

- [ ] slide vertically on the right side to change volume
- [ ] slide vertically on the left side to change brightness
- [ ] Pinch-to-Zoom to toggle between full video (with borders) or fill screen (and cut edges)
- [ ] sliding horizontally to seek
- [ ] double tapping to toggle play/pause,

**PS:** [Here's a link to documentation](https://developer.android.com/training/gestures/detector) in case someone wants to give it a try

**Comment by Pentaphon** [comment]
>Pinch-to-Zoom to toggle between full video (with borders) or fill screen (and cut edges)
>double tapping to toggle play/pause,

I don't think we need these gestures. Pinch to zoom isn't as obvious to most people as the fit button, and togging play/pause with a double tap is way more annoying and more taps than just tapping the play/pause button. 

The rest are necessary, especially if we want to remove that HUGE side volume display which makes the player look outdated. 

I also think gestures should have a nice animated OSD indicator as well to make the player look more modern.

**Comment by alexanderadam** [comment]
> > Pinch-to-Zoom to toggle between full video (with borders) or fill screen (and cut edges)
> > double tapping to toggle play/pause,
> 
> I don't think we need these gestures.

I mean, we obviously don't need gestures at all, since they are not necessary to watch anything. It's just about UX improvements. :wink: 
But I agree that someone might want to have gestures switchable in the settings.
Just recently discovered it by accident that YouTube or Vanced have this feature and since then I love it to make full usage of my tablet screen.

> I also think gestures should have a nice animated OSD indicator as well to make the player look more modern.

:+1:

**Comment by onthejobtraining** [comment]
> togging play/pause with a double tap is way more annoying and more taps than just tapping the play/pause button.

I disagree, double tapping to play/pause is easier imo because the play/pause doesn't appear all the time. First we need to touch somewhere on the screen to make the button appear and than reach for the button. Whereas, double tapping can be done from literally anywhere on the screen. So it's just more convenient, especially for lazy bums like me.

Edit: There's also MX Player Pro that allows double tap to be disabled, or use it for either Play/Pause or Rewind/Skip.
Nova can take away a lot from MXP Pro and up its gestures game imo.

**Comment by Ilithy** [comment]
As gestures to possibly add there are the triple tap on the left and right edge to move backwards and forwards slightly, and the forward is configurable
_(VLC uses it, I like this option)_

Thanks

**Comment by courville** [comment]
OK let's give it a try.

**Comment by Pentaphon** [comment]
> OK let's give it a try.

Will you also remove that volume overlay on the left side of the screen and instead put in a volume gesture when using the device on a touchscreen? I think Just Player or Newpipe do gestures pretty well if you need open source examples.

**Comment by courville** [comment]
> > OK let's give it a try.
> 
> Will you also remove that volume overlay on the left side of the screen and instead put in a volume gesture when using the device on a touchscreen? I think Just Player or Newpipe do gestures pretty well if you need open source examples.

For now old UI is still available on single touch. Will have a try to get a more modern look soon.

**Comment by courville** [comment]
Following gestures are now supported:
- double tap on left/right regions of the screen for +-10s seek backward/forward
- scroll on left/right regions of the screen for brightness/volume increase/decrease

Two osd regions have been added on left/right screen part to display the frew/ffwd icons and volume/brightness levels.

**Comment by Pentaphon** [comment]
Edit: still open

**Comment by vertigo220** [comment]
> @courville thanks for the .38 update. It was long overdue on touchscreens. This can be closed.

This is not done, yet, and can't/shouldn't be closed. It's great the volume and brightness swipe gestures have been added, as that was one of the reasons I preferred other players, same with double-tapping to go forward/backward, but IMO it still very much needs a horizontal swipe to seek, which is not only more intuitive (I tried that and was dismayed to find it missing, whereas I didn't think to try double-tapping) but also easier to use, especially to seek more than half a minute or so.

**Comment by Pentaphon** [comment]
> IMO it still very much needs a horizontal swipe to seek, which is not only more intuitive (I tried that and was dismayed to find it missing, whereas I didn't think to try double-tapping) but also easier to use, especially to seek more than half a minute or so.

That's true. You did ask for that. Never mind my post then. I'm sure courville will give it a try.

**Comment by Korb** [comment]
> it still very much needs a horizontal swipe to seek, which is not only more intuitive (...) but also easier to use, especially to seek more than half a minute or so.

I completely agree with this statement. Considering that the **horizontal swipe** is not currently used in the application, it is strange not to use this UX element, which is familiar to other products. I understand situations where an application is developed for all mobile devices, including e-ink screens. There, for a number of reasons, they try to avoid gestures in favor of traditional _point presses_. But this is not the case here, and it is strange not to use the historically established management standard.

> Pinch-to-Zoom to toggle between full video (with borders) or fill screen (and cut edges)

I also miss this feature, which allows you to slightly enlarge the video by cropping the top and bottom, while maintaining the aspect ratio, but not to the full width of the screen.

**Comment by eng3** [comment]
I will add that pinch and zoom would be a very useful feature even my 2-year-old niece knows how to pinch into even if it's not a touch screen.  With videos being in different formats 16x9 4x32.35x1 etc and some having black bars and phone screens being different sizes it is often better to have the ability to pinch consume to be more precise.  Also sometimes you just want to zoom in on a video because it's not framed very well
